npm/npm_import.bzl

```diff
@@ -313,24 +373,40 @@ def npm_translate_lock(
             ```
             lifecycle_hooks_envs: {
                 "*": ["GLOBAL_KEY1=value1", "GLOBAL_KEY2=value2"],
-                "@foo/bar": ["PREBULT_BINARY=http://downloadurl"],
+                "@foo/bar": ["GLOBAL_KEY2=", "PREBULT_BINARY=http://downloadurl"],
             }
             ```
 
+            Execution variables are additive when there are multiple matches for a package. More specific matches are
+            appended to previous matches.
+
+            In the above example, `@foo/bar` will have the environment variables
+            `["GLOBAL_KEY1=value1", "GLOBAL_KEY2=value2", "GLOBAL_KEY2=", "PREBULT_BINARY=http://downloadurl"]` set where
+            the 2nd `GLOBAL_KEY` environment value will override the first.
+
         lifecycle_hooks_execution_requirements: Execution requirements applied to the preinstall, install and postinstall
             lifecycle hooks on npm packages.
 
-            The execution requirements can be defined per package by package name or globally using "*".
-
-            For example:
+            The execution requirements can be defined per package by package name or globally using "*". For example,
 
             ```
-            lifecycle_hooks_execution_requirements: {
-                "*": ["requires-network"],
-                "@foo/bar": ["no-sandbox"],
+            lifecycle_hooks_execution_requirements = {
+                "*": ["no-sandbox"],
+                "@foo/bar": [],
+                "fum@0.0.1": ["no-sandbox", "no-remote-exec"],
             }
             ```
 
+            Execution requirements are not additive. The move specific name match takes precedence.
+
+        lifecycle_hooks_no_sandbox: If True, a "no-sandbox" execution requirement is added to all lifecycle hooks
```

Comment:
want to deprecate this one?

Reply:
maybe in a follow-up

Reply:
Note there's already a TODO to remove it in 2.0.. 🤷

Reply:
The TODO is for the one in `npm_import`. That one is rarely used and is not an API that is used by most users.

The one in `npm_translate_lock` I'm not sure what to do about yet so I'll punt the decision to a follow-up.